### PR TITLE
Expose block ID in SVG.

### PR DESCRIPTION
**Pull request by NeilFraser**
Useful for debugging.

### CODE REVIEW

File: core/block_svg.js [hunk 1]
```diff
@@ -99,6 +99,9 @@ Blockly.BlockSvg = function(workspace, prototypeName, opt_id) {
   Blockly.Tooltip.bindMouseEvents(this.svgPath_);
   Blockly.BlockSvg.superClass_.constructor.call(this,
       workspace, prototypeName, opt_id);
+
+  // Expose this block's ID on its top-level SVG group.
+  this.svgGroup_.id = this.id;
```
Comment (AnmAtAnm) [inline core/block_svg.js]: Don't these XML ids have strong potential for duplication in the document.  Consider a page with toolbox XML embedded in the HTML. There would be two elements with the same id: the toolbox `<block>` element in the XML and this new `<svg:g>` element.

I suggest using a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes): `svgGroup_.dataset.blockId = this.id;`

Comment (NeilFraser) [reply core/block_svg.js]: Alternatively, prefix the SVG ID with 'block'.  Then it becomes clearer when browsing the DOM what that group actually is.

Comment (NeilFraser) [reply core/block_svg.js]: Here's a solution that also addresses the issue that currently there's no way at all to associate a bubble with it's block by inspecting the DOM: https://github.com/google/blockly/pull/1700

Comment (AnmAtAnm) [reply core/block_svg.js]: Testing with the playground (adding the id to the a block), this still isn't sufficient. If a toolbox block has an id, than the first copy of the that block in the workspace has the same id. Thus, the prefix differentiates the `<svg:g>` from the toolbox `<block>`, but does not uniquely identify the element in the document.

Looks like you may already be making this change in #1700.

**Pull request merged**